client/boot/index.js

```diff
@@ -228,6 +231,26 @@ function reduxStoreReady( reduxStore ) {
 	// Layout themselves when logged in.
 	if ( ! isIsomorphic || user.get() ) {
 		renderLayout( reduxStore );
+
+		if ( config.isEnabled( 'catch-js-errors' ) ) {
+			const Logger = require( 'lib/catch-js-errors' );
+			const errorLogger = new Logger();
+			//Save data to JS error logger
+			errorLogger.saveDiagnosticData( {
+				user_id: user.get().ID,
+				calypso_env: config( 'env_id' )
+			} );
+			errorLogger.saveDiagnosticReducer( function() {
+				const state = reduxStore.getState();
+				return {
+					blog_id: getSelectedSiteId( state ),
+					calypso_section: getSectionName( state )
+				};
+			} );
+			errorLogger.saveDiagnosticReducer( () => ( { tests: getSavedVariations() } ) );
+			analytics.on( 'record-event', ( eventName, eventProperties ) => errorLogger.saveExtraData( { lastTracksEvent: eventProperties } ) );
+			analytics.on( 'page-view', ( urlPath ) => errorLogger.saveDiagnosticData( { calypso_path: urlPath } ) );
```

Comment:
I don't think we should be watching for the path from our analytics module. Instead we should tie into the `page.js` instance, and we should update the path at the start of the page event. At that point you will have the _intended_ path before we use pushState to update the URL as shown in the browser. I believe analytics will only be called after this has happened, so you'll still be one step too late to catch the path that has caused the problem.